BridgeAppSDK/SBAProfileItem.swift

```diff
@@ -523,7 +557,170 @@ open class SBAUserDefaultsProfileItem: SBAProfileItemBase {
 
 }
 
-open class SBAFullNameProfileItem: SBAKeychainProfileItem, SBANameDataSource {
+enum SBAProfileParticipantSourceKey: String {
+    case firstName
+    case lastName
+    case email
+    case externalId
+    case notifyByEmail
+    case sharingScope
+    case dataGroups
+}
+
+open class SBAStudyParticipantProfileItem: SBAStudyParticipantCustomAttributesProfileItem {
+    public static var studyParticipant: SBBStudyParticipant?
+    
+    override open func storedValue(forKey key: String) -> Any? {
+        guard let studyParticipant = SBAStudyParticipantProfileItem.studyParticipant
+            else {
+                assertionFailure("Attempting to read \(key) (\(profileKey)) on nil SBBStudyParticipant")
+                return nil
+        }
+        guard !(key.hasPrefix("attributes.")) else {
+            let pathComponents = key.components(separatedBy: ".")
+            let attributeComponents = pathComponents[1...pathComponents.count - 1]
```

Comment:
`let attributeComponents = pathComponents[1..<pathComponents.count]` is equivalent and reads a bit easier.